test/api/peer.transactions.transfer.js

```diff
@@ -0,0 +1,177 @@
+			expect(err).to.not.exist;
+			node.onNewBlock(done);
+		});
+	});
+
+	it('should return error when data field length is greater than 64 characters', function (done) {
+		var data = new Array(65).fill('x').join('');
+		var trs = node.lisk.transaction.createTransaction(node.randomAccount().address, 1000, account.password, null, data);
+		addTransaction(trs, function (err, res) {
+			expect(err).to.not.exist;
+			expect(res.success).to.equal(false);
+			expect(res.message).to.equal('Invalid transaction body - Failed to validate transfer schema: String is too long (65 chars), maximum 64');
+			done();
+		});
+	});
+
+	it('should return error when data field is not a string', function (done) {
+		var data = 0;
+		var trs = node.lisk.transaction.createTransaction(node.randomAccount().address, 1000, account.password, null, data);
+		trs.asset.data = 0;
+		addTransaction(trs, function (err, res) {
+			expect(err).to.not.exist;
+			expect(res.success).to.equal(false);
+			expect(res.message).to.equal('Invalid transaction body - Failed to validate transfer schema: Expected type string but found type integer');
+			done();
+		});
+	});
+
+	it('should create transaction with empty asset', function (done) {
+		var trs = node.lisk.transaction.createTransaction(node.randomAccount().address, 1000, account.password);
+		addTransaction(trs, function (err, res) {
+			expect(err).to.not.exist;
+			var transactionId = res.transactionId;
+			node.onNewBlock(function (err) {
+				getTransactionById(transactionId, function (err, res) {
+					expect(err).to.not.exist;
+
+					expect(res).to.have.property('transaction').which.is.an('object');
+					expect(res.transaction.id).to.equal(trs.id);
+					expect(res.transaction.amount).to.equal(trs.amount);
+					expect(res.transaction.asset).to.eql({});
+					expect(res.transaction.fee).to.equal(trs.fee);
+					expect(res.transaction.type).to.equal(trs.type);
+					done();
+				});
+			});
+		});
+	});
+
+	it('should return error when data field characters length is equal to 0', function (done) {
+		var trs = node.lisk.transaction.createTransaction(node.randomAccount().address, 1000, account.password, null);
+		trs.asset.data = '';
+		addTransaction(trs, function (err, res) {
+			expect(err).to.not.exist;
+			expect(res.success).to.equal(false);
+			expect(res.message).to.equal('Invalid transaction body - Failed to validate transfer schema: String is too short (0 chars), minimum 1');
+			done();
+		});
+	});
+
+	it('should be able to create transaction with 0 value', function (done) {
+		var data = '0';
+		var trs = node.lisk.transaction.createTransaction(node.randomAccount().address, 1000, account.password, null, data);
+		addTransaction(trs, function (err, res) {
+			expect(err).to.not.exist;
+			var transactionId = res.transactionId;
+			node.onNewBlock(function (err) {
+				getTransactionById(transactionId, function (err, res) {
+					expect(err).to.not.exist;
+
+					expect(err).to.not.exist;
```

Comment:
Why duplicate expectations on `err`? Repeated below also in subsequent examples.